src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -105,10 +113,37 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
         Address updatedAddress = editPersonDescriptor.getAddress().orElse(personToEdit.getAddress());
         Course updatedCourse = editPersonDescriptor.getCourse().orElse(personToEdit.getCourse());
         Tag updatedTag = editPersonDescriptor.getTag().orElse(personToEdit.getTag());
+
         ArrayList<Module> updatedModules = editPersonDescriptor.getModules().orElse(personToEdit.getModules());
+        if (editPersonDescriptor.hasModuleChanges()) {
+            Module oldModule = editPersonDescriptor.oldModule;
+            Module newModule = editPersonDescriptor.newModule;
+
+            boolean isModuleRenamed = false;
+            if (updatedModules.stream().anyMatch(m -> m.value.equals(newModule.value))) {
+                throw new CommandException(EditCommand.MESSAGE_DUPLICATE_MODULE);
+            }
+
+            for (int i = 0; i < updatedModules.size(); i++) {
```

Comment:
Good naming choices: oldModule, newModule, and updatedModules, clearly distinguishing between singular and plural object, improving code readability and understanding